test/unity/toolkits/object_detection/test_object_detector.cxx

```diff
@@ -445,6 +447,63 @@ BOOST_AUTO_TEST_CASE(test_object_detector_train) {
 
   // Deconstructing `model` here will assert that every expected call to a
   // mocked-out method has been called.
+
+  // Test Non Maximum Suppression model
+  auto test_iou_threshold = 0.55;
+  auto test_confidence_threshold = 0.15;
+    
+  std::map<std::string, flexible_type> options;
+  options["include_non_maximum_suppression"] = 1;
+  options["iou_threshold"] = test_iou_threshold;
+  options["confidence_threshold"] = test_confidence_threshold;
+  
+  flex_dict user_defined_metadata;
+  user_defined_metadata.emplace_back("model", "model");
+  user_defined_metadata.emplace_back("max_iterations", test_max_iterations);
+  user_defined_metadata.emplace_back("training_iterations", test_max_iterations);
+  user_defined_metadata.emplace_back("include_non_maximum_suppression", "True");
+  user_defined_metadata.emplace_back("feature", test_image_name);
+  user_defined_metadata.emplace_back("annotations", test_annotations_name);
+  user_defined_metadata.emplace_back("classes", "label1, label2");
+  user_defined_metadata.emplace_back("type", "object_detector");
+  user_defined_metadata.emplace_back("confidence_threshold", options["confidence_threshold"]);
+  user_defined_metadata.emplace_back("iou_threshold", options["iou_threshold"]);
+    
+  flex_list t_class_labels = flex_list(test_class_labels.begin(), test_class_labels.end());
```

Comment:
Could we test that the output includes these labels?

Reply:
Added. Thanks.